doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
+
++ Request (application/x-www-form-urlencoded)
+    + Attributes
+        + username: the_username (string, required)
+        + password: aSecretPassword (string, required)
```

Comment:
> In this case I can only think of using HTTPS instead.

This is also what I expect to be the most promising protection, but it doesn't really impact my work I think - it is only a configuration task later on.

> But challenge/response offers better protection of the password itself than plaintext or hashing.

I totally agree, but implementing a challenge/response system within HTTP is kinda akward. It may work quite well if you have an API and a frontend that takes care of, but when using the API by hand (curl), I can't imagine how it should work. My approach was also to not store states on the server (that's why I'm using stateless JWTs), because it makes development a lot more complex and error prone.